### PR TITLE
feat: allow increased testnet deposits

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Changes deposit command to rely on endpoint from https://github.com/iron-fish/ironfish-api/pull/1053

Defaults to max deposit amount, if not available in account, defaults to account balance - fee
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

**Comment by hairtail** [comment]
Hi, does `minDeposit` and `maxDeposit` change during testnet? If not, it's better to use `minDeposit = 10` because some miners can be stuck for network reason. You know, the api server is based in USA.

File: ironfish-cli/src/commands/deposit.ts
Comment (jowparks) [inline ironfish-cli/src/commands/deposit.ts:128]: Is it acceptable to default to spendable balance (total wallet balance - fee)? Or should we be doing something else?

**Comment by jowparks** [comment]
> Hi, does `minDeposit` and `maxDeposit` change during testnet? If not, it's better to use `minDeposit = 10` because some miners can be stuck for network reason. You know, the api server is based in USA.

Yes the intention of this PR is to unblock users from depositing their Iron for getting their points during testnet. The current deposit limit is 0.1 Iron, waiting on some of the other devs to comment on what the correct default value is.

File: ironfish/src/webApi.ts
Comment (jowparks) [inline ironfish/src/webApi.ts:128]: Before merging, we should at minimum test this endpoint.

Comment (jowparks) [reply ironfish/src/webApi.ts:128]: Endpoint is in staging: https://ironfish-api-staging.herokuapp.com/deposits/min_and_max_deposit_size

File: ironfish-cli/src/commands/deposit.ts
Comment (jowparks) [inline ironfish-cli/src/commands/deposit.ts:121]: Should we convert this `fee` log to IRON? Seemed like it would be a bit extreme to put in IRON (would be like 0.0000000001 IRON, right?)

Comment (NullSoldier) [reply ironfish-cli/src/commands/deposit.ts:121]: I think so, yes. IRON at the edges, ore internally.

On a side note, I think the language could be reduced down a bit. Most of our user's do not speak english as a first language.

**Review by NullSoldier — APPROVED** [review]

File: ironfish-cli/src/commands/deposit.ts
Comment (jowparks) [reply ironfish-cli/src/commands/deposit.ts:121]: Good thinking, I didn't even consider that. I will change

File: ironfish-cli/src/commands/deposit.ts
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:132]:
```suggestion
        )} IRON.  Fee (${CurrencyUtils.renderIron(
```

File: ironfish-cli/src/commands/deposit.ts
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:134]:
```suggestion
        )} IRON) + minimum deposit (${minDepositSize} IRON) = total required (${CurrencyUtils.renderIron(
```

File: ironfish-cli/src/commands/deposit.ts
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:146]: we can delete this now

File: ironfish-cli/src/commands/deposit.ts
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:204]:
```suggestion
            amount: CurrencyUtils.encode(oreToSend),
```

**Review by NullSoldier — APPROVED** [review]

File: ironfish/src/webApi.ts
Comment (NullSoldier) [reply ironfish/src/webApi.ts:128]: 👍

**Pull request merged**
